wasm-runtime/polkadot/src/support/storage.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright 2017 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Stuff to do with the runtime's storage.
+
+use runtime_std::prelude::*;
+use runtime_std::{self, twox_128};
+use codec::{Slicable, KeyedVec};
+
+// TODO: consider using blake256 to avoid possible preimage attack.
+
+/// Return the value of the item in storage under `key`, or `None` if there is no explicit entry.
+pub fn get<T: Slicable + Sized>(key: &[u8]) -> Option<T> {
+	Slicable::set_as_slice(&|out, offset|
+		runtime_std::read_storage(&twox_128(key)[..], out, offset) >= out.len()
+	)
+}
+
+/// Return the value of the item in storage under `key`, or the type's default if there is no
+/// explicit entry.
+pub fn get_default<T: Slicable + Sized + Default>(key: &[u8]) -> T {
```

Comment:
Is there much of an advantage to having this function over having users call

`storage::get(key).unwrap_or_default()`?

Reply:
didn't realise that was stable. but i don't think it's confusing and it's convenient.